backend/app/api/events.py

```diff
@@ -54,10 +56,78 @@ def query_eventbrite(city: str, date: str, keywords: List[str]) -> List[Event]:
     
     return events
 
+def get_formatted_date(date_entity): 
+    current_date = datetime.now() 
+    # Handle relative dates like "today" or "tomorrow"
+    if date_entity:
+        date_entity = date_entity.lower()
+        if "today" in date_entity:
+            return current_date.replace(hour=0, minute=0, second=0, microsecond=0).isoformat(), current_date.replace(hour=23, minute=59, second=59, microsecond=0).isoformat()
+        elif "tomorrow" in date_entity:
+            start = current_date.replace(hour=0, minute=0, second=0, microsecond=0)+ timedelta(days=1)
+            end = current_date.replace(hour=23, minute=59, second=59, microsecond=0)+ timedelta(days=1)
+            return start.isoformat(), end.isoformat()
+        else:
+            # Handle YYYY-MM-DD format
+            try:
+                parsed_date = datetime.strptime(date_entity, "%Y-%m-%d")
+                start_date = parsed_date.replace(hour=0, minute=0, second=0, microsecond=0)
+                end_date = parsed_date.replace(hour=23, minute=59, second=59, microsecond=0)
+                return start_date.isoformat(), end_date.isoformat()
+            except ValueError:
+                # Handle specific formats like "23-11" (dd-mm)
+                try:
+                    parsed_date = datetime.strptime(date_entity, "%d-%m").replace(year=current_date.year)
+                    start_date = parsed_date.replace(hour=0, minute=0, second=0, microsecond=0)
+                    end_date = parsed_date.replace(hour=23, minute=59, second=59, microsecond=0)
+                    return start_date.isoformat(), end_date.isoformat()
+                except ValueError:
+                    return None, None
+                
+def query_ticketmaster(city: str, date: str, keywords: List[str]) -> List[Event]:
+    """Fetch events from Ticketmaster API."""
+    # Remove the date and city from keywords to avoid confusion using API
+    filtered_keywords = [kw for kw in keywords if kw.lower() != city.lower() and kw.lower() != date.lower()]
+    # Set the day start and end, handle cases like "tomorrow", "today"
+    date, end_date = get_formatted_date(date)
+    print(f"Querying Ticketmaster: city={city}, date={date}, endDate={end_date}, keywords={filtered_keywords}")
+
+    params = {
+        "keyword": " ".join(filtered_keywords),
+        "city": f"{city}",
+        "sort": "date,asc",
+        "startDateTime": date + "+00:00",
```

Comment:
What if `date` and `end_date` are `None`?

Reply:
they wouldn't be cuz the chat screens warns the user if no date is entered
```
 city = info.get('city')
    date = info.get('date')
    keywords = info.get('keywords')
    if city is None or date is None:
        CITY_NOT_FOUND = "You seem to have not provided the city or date correctly. Please double check it"
```